### PR TITLE
 core/ondisk: faster read_varint 

### PR DESCRIPTION
**Pull request by pereman2**
```
Insert rows in batches/read_varint
                        time:   [402.67 ns 408.01 ns 417.14 ns]
Insert rows in batches/read_varint_slow
                        time:   [527.01 ns 531.45 ns 535.57 ns]
```

### CODE REVIEW

File: core/benches/benchmark.rs [hunk 1]
```diff
@@ -598,9 +598,37 @@ fn bench_insert_rows(criterion: &mut Criterion) {
     group.finish();
 }
 
+fn bench_read_varint(criterion: &mut Criterion) {
+    let mut group = criterion.benchmark_group("Insert rows in batches");
```
Comment (sonhmai) [inline core/benches/benchmark.rs:602]: typo? `benchmark_group("Read varint")`

File: core/storage/sqlite3_ondisk.rs [hunk 2]
```diff
@@ -2208,4 +2348,31 @@ mod tests {
 
         assert_eq!(small_vec.get(8), None);
     }
+
+    #[test]
+    fn test_read_varint() {
```
Comment (sonhmai) [inline core/storage/sqlite3_ondisk.rs:2353]: I notice in edge cases e.g. where we pass in invalid buf like `&[]` the slow version returns an Error while the new version panics with index out of bound. Is it a concern here?

Maybe adding tests with error cases would help catch those discrepancies.

**Comment by sonhmai** [comment]
While reading through this, I also noticed missing/ extra bytes in the buf like `&[0x80]` cause panic as well. Bound checks in general was not present in the original slow version. In case bound check for edge cases in general is needed, leaving a note here so it can be introduced it in another issue/ PR.

**Comment by pereman2** [comment]
@sonhmai thanks for noticing I will fix promptly

**Comment by pereman2** [comment]
I've decided to close this after not seeing clear evidence of better performance. I added more varints to the benchmark data and the results were meh:
```
Read varint/read_varint_slow
                        time:   [18.879 µs 19.008 µs 19.154 µs]
                        change: [-0.0110% +1.2404% +2.5617%] (p = 0.06 > 0.05)
                        No change in performance detected.
Found 6 outliers among 100 measurements (6.00%)
  6 (6.00%) high mild
Read varint/read_varint time:   [20.640 µs 20.785 µs 20.944 µs]
                        change: [-2.3078% -1.3711% -0.4215%] (p = 0.00 < 0.05)
                        Change within noise threshold.
                        ```
kinda sad.
